pkg/executor/api.go

```diff
@@ -249,6 +249,62 @@ func (executor *Executor) unTapService(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+// isValidHandler is very similar to getServiceForFunctionAPI, with the difference that the response returned doens't contain
+// the address, but instead returns "true" or "false", depending if the function service is valid or not.
+func (executor *Executor) isValidHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := otelUtils.LoggerWithTraceID(ctx, executor.logger)
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request", http.StatusInternalServerError)
+		return
+	}
+
+	// get function metadata
+	fn := &functionWithAddress{}
+	err = json.Unmarshal(body, &fn)
+	if err != nil {
+		logger.Error("Error unmarshelling: ", zap.Any("error", err))
+		http.Error(w, "Failed to parse request", http.StatusBadRequest)
+		return
+	}
+
+	t := fn.Function.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType
+	et := executor.executorTypes[t]
+	funct := fn.Function
+
+	if t == fv1.ExecutorTypePoolmgr && !funct.Spec.OnceOnly {
+		fsvc, err := et.GetFuncSvcFromPoolCacheByAddress(ctx, &fn.Function, fn.SvcAddress)
+
+		// check if its a cache hit (check if there is already specialized function pod that can serve another request)
+		if err == nil {
+			if et.IsValid(ctx, fsvc) {
+				// return true
```

Comment:
These comments are not helpful, they describe **_what_** the code is doing rather than **_why_** the code is doing it. Also the amount of whitespace here is a bit excessive. I think the code / log messages are easy to follow at this point and we probably don't need any comments in these blocks

Reply:
I agree. Just changed it. Please check it out!